src/proto/code.rs

```diff
@@ -0,0 +1,84 @@
+use super::error::{ProtoError, ProtoResult};
+
+/// Message numbers used for requests from the client to the agent.
+///
+/// https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent#name-message-numbers
+#[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord)]
+#[allow(missing_docs)]
+#[repr(u8)]
+pub enum RequestCode {
+    RequestIdentities = 11,
+    SignRequest = 13,
+    AddIdentity = 17,
+    RemoveIdentity = 18,
+    RemoveAllIdentities = 19,
+    AddSmartcardKey = 20,
+    RemoveSmartcardKey = 21,
+    Lock = 22,
+    Unlock = 23,
+    AddIdConstrained = 25,
+    AddSmartcardKeyConstrained = 26,
+    Extension = 27,
```

Comment:
In the spec, [SSH agent protocol numbers]](https://datatracker.ietf.org/doc/html/draft-miller-ssh-agent#section-6.1) carves out `240-255` as "reserved for organizational use".

Is this worth including? Noting though that this may be a bit messy (given they could be requests or responses).

Though, another way to handle this case is just to let the codec handle the error and perform the resolution itself

Reply:
Yet another way would be to add a catch-all variant like so: https://github.com/rpgp/rpgp/blob/master/src/crypto/hash.rs#L31-L32